src/link/language-selector.js

```diff
@@ -37,12 +46,17 @@ export const LanguageSelector = ( { setLanguageSelector, setLang } ) => {
 
 		const filtered = Object.keys( languages )
 			.filter( ( language ) => {
+				const localized =
+					language.indexOf( '-' ) === -1 &&
```

Comment:
in the example given in mdn, still possible the language has `-`

```js
// Get display names of language in English
let languageNames = new Intl.DisplayNames(['en'], {type: 'language'});
languageNames.of('fr');      // "French"
languageNames.of('de');      // "German"
languageNames.of('fr-CA');   // "Canadian French"
languageNames.of('zh-Hant'); // "Traditional Chinese"
languageNames.of('en-US');   // "American English"
languageNames.of('zh-TW');   // "Chinese (Taiwan)"]
```

Reply:
You're right, it turns out there are a certain number of languages which make the `localizedName.of( language )` throw with an error. The invalid languages and error are:

```
['bat-smg', 'be-x-old', 'cbk-zam', 'fiu-vro', 'map-bms', 'roa-rup', 'zh-classical', 'zh-min-nan', 'zh-yue', 'zh-cdo'];
```

```
Uncaught RangeError: invalid value "bat-smg" for option language
    getLocalized language-selector.js:35
    filtered language-selector.js:52
    filterLanguages language-selector.js:48
    onChange components.min.js:10
```

It would be really nice to have some type of built-in support with `Intl` that would let us know if a language is invalid before hand. I looked into [supportedLocalesOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames/supportedLocalesOf) and [resolvedOptions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames/resolvedOptions) for example, but no luck. One solution is selectively skipping these 10 edge cases, which my last commit shows, though I'm curious to know if there's a better way.

Reply:
is that possible there will be more and how do you get these 10 cases? 
If you can confirm then I don't see any other solution.

Reply:
I narrowed it down to these 10 cases manually by testing, but I don't know what's the probability of new "invalid" languages being added in the future. Looking at the [source commit history](https://github.com/mdn/content/commits/main/files/en-us/web/javascript/reference/global_objects/intl/displaynames), it's about 1-2 commits per months it seems. I checked the listed [issues](https://github.com/mdn/content/issues) also, nothing stands out.